In [1]:
!pip install wikipedia
!pip install folium
!pip install geopy
!pip install geopandas
!pip install geojson
!pip install OSMPythonTools
!pip install reverse_geocoder

In [2]:
import pandas as pd
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe
from geopy.extra.rate_limiter import RateLimiter
import matplotlib.pyplot as plt
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import wikipedia as wp
import requests
from bs4 import BeautifulSoup
import numpy as np # library to handle data in a vectorized manner
import json # library to handle JSON files
from geopy.geocoders import Nominatim # convert an address into latitude and longitude values
import geopandas as gpd

# Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors
import matplotlib.pyplot as plt

import folium # map rendering library
from folium import plugins
from folium.plugins import HeatMap

print('Libraries imported.')

Libraries imported.


In [3]:
df= pd.read_html('https://de.wikipedia.org/wiki/Liste_der_Ortsteile_von_Erfurt', header=0)
Erfurt_borough_wiki = df[0]

Erfurt_borough_wiki.columns = ('district', 'area1', 'population', 'km2', 'Population_Density_km2', 'structur', 'year', 'district2', 'houses', 'flats', 'empty')
Erfurt_borough_wiki

district  area1  population   km2  Population_Density_km2  \
0                Alach    NaN       1.000   101                  99.000   
1             Altstadt    NaN      17.980   244                   7.369   
2      Andreasvorstadt    NaN      16.020   285                   5.621   
3          Azmannsdorf    NaN     339.000   425                  80.000   
4       Berliner Platz    NaN       5.844    48                  12.175   
5         Bindersleben    NaN       1.317   806                 163.000   
6   Bischleben-Stedten    NaN       1.643   661                 249.000   
7      Brühlervorstadt    NaN      12.870   708                   1.818   
8             Büßleben    NaN       1.320   685                 193.000   
9           Daberstedt    NaN      13.488    35                   3.854   
10         Dittelstedt    NaN     793.000   194                 409.000   
11             Egstedt    NaN     496.000  1257                  39.000   
12            Ermstedt    NaN     449.000   587                  76.000   
13          Frienstedt    NaN       1.343   719                 187.000   
14        Gispersleben    NaN       4.107  1014                 405.000   
15           Gottstedt    NaN     215.000    21                 102.000   
16          Herrenberg    NaN       7.937   169                   4.696   
17            Hochheim    NaN       2.665   284                 938.000   
18           Hochstedt    NaN     271.000   297                  91.000   
19         Hohenwinden    NaN       1.902   817                 233.000   
20       Ilversgehofen    NaN      11.202   274                   4.088   
21       Johannesplatz    NaN       5.172    43                  12.028   
22    Johannesvorstadt    NaN       6.123   330                   1.855   
23          Kerspleben    NaN       1.748  1043                 168.000   
24    Krämpfervorstadt    NaN      15.601   496                   3.145   
25          Kühnhausen    NaN       1.165   314                 371.000   
26          Linderbach    NaN     833.000   312                 267.000   
27       Löbervorstadt    NaN      11.991  1024                   1.171   
28             Marbach    NaN       3.821   514                 743.000   
29         Melchendorf    NaN      10.234   563                   1.818   
30        Mittelhausen    NaN       1.084  1047                 104.000   
31     Möbisburg-Rhoda    NaN       1.098   803                 137.000   
32            Molsdorf    NaN     557.000   717                  78.000   
33      Moskauer Platz    NaN       7.638    88                   8.680   
34         Niedernissa    NaN       1.578   379                 416.000   
35               Rieth    NaN       5.650    72                   7.847   
36    Rohda (Haarberg)    NaN     279.000   360                  78.000   
37          Roter Berg    NaN       5.691    77                   7.391   
38        Salomonsborn    NaN       1.081   468                 231.000   
39          Schaderode    NaN     294.000   142                 207.000   
40             Schmira    NaN     947.000   727                 130.000   
41          Schwerborn    NaN     605.000   691                  88.000   
42        Stotternheim    NaN       3.506  1575                 223.000   
43     Sulzer Siedlung    NaN       1.040    96                   1.083   
44            Tiefthal    NaN       1.086   523                 208.000   
45         Töttelstädt    NaN     632.000  1079                  59.000   
46           Töttleben    NaN     324.000   334                  97.000   
47              Urbich    NaN       1.106   285                 388.000   
48          Vieselbach    NaN       2.194   772                 284.000   
49           Wallichen    NaN     168.000   237                  71.000   
50        Waltersleben    NaN     433.000   484                  89.000   
51         Wiesenhügel    NaN       5.112    45                  11.360   
52  Windischholzhausen    NaN       1.6

In [4]:
Erfurt_district = Erfurt_borough_wiki.drop(['area1', 'population', 'km2', 'Population_Density_km2', 'structur', 'year', 'district2', 'houses', 'flats', 'empty'],axis=1)
Erfurt_district.head()

district
0            Alach
1         Altstadt
2  Andreasvorstadt
3      Azmannsdorf
4   Berliner Platz

In [5]:
address = 'Erfurt, Thuringia'

geolocator = Nominatim(user_agent="Erfurt")
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinates of Erfurt are {}, {}.'.format(latitude, longitude))

The geograpical coordinates of Erfurt are 50.9777974, 11.0287364.


In [6]:
geolocator = Nominatim(user_agent="Erfurt")

list_latitude = []
list_longitude = []

for suburb in Erfurt_district["district"]:
    suburb = suburb
    town = "Erfurt"
    loc = geolocator.geocode(suburb+','+ town)
    list_latitude.append(loc.latitude)
    list_longitude.append(loc.longitude)

In [7]:
Erfurt_district['Latitude']=list_latitude
Erfurt_district['Longitude']=list_longitude
Erfurt_district.head()

district   Latitude  Longitude
0            Alach  50.987442  10.928551
1         Altstadt  50.979617  11.023882
2  Andreasvorstadt  50.988127  11.012060
3      Azmannsdorf  50.989126  11.107705
4   Berliner Platz  51.004539  11.005643

In [8]:

import types
import pandas as pd
from botocore.client import Config
import ibm_boto3

def __iter__(self): return 0

# @hidden_cell
# The following code accesses a file in your IBM Cloud Object Storage. It includes your credentials.
# You might want to remove those credentials before you share the notebook.
client_90e381095a1e4558b2df2cd2d4d19226 = ibm_boto3.client(service_name='s3',
    ibm_api_key_id='5-zqmKt3KE7JbkF0f_M1cj8OwK9jS1g2Y8oKyD7WV9R5',
    ibm_auth_endpoint="https://iam.cloud.ibm.com/oidc/token",
    config=Config(signature_version='oauth'),
    endpoint_url='https://s3.eu-geo.objectstorage.service.networklayer.com')

body = client_90e381095a1e4558b2df2cd2d4d19226.get_object(Bucket='capstoneprojectthebattleofneighbo-donotdelete-pr-mntqtfvn6tn9cv',Key='Nursing_and_care_home.csv')['Body']
# add missing __iter__ method, so pandas accepts body as file-like object
if not hasattr(body, "__iter__"): body.__iter__ = types.MethodType( __iter__, body )
  
    #index_col =["name","street","house_number","postalcode","city"]
Nursing_and_care_home = pd.read_csv(body,sep=';', header=0, index_col=False) 
Nursing_and_care_home

name  \
0                              Helios Klinikum Erfurt   
1   Katholisches Krankenhaus "St. Johann Nepomuk" ...   
2        Alloheim Senioren-Residenz "Am Hirschgarten"   
3          ASB Senioren- und Pflegeheim "Georg Boock"   
4        Augusta-Viktoria-Stift Alten- und Pflegeheim   
5                   Augusta-Viktoria-Stift Pflegeheim   
6   AWO-Altenpflegeheim Haus zu den vier Jahreszeiten   
7                       AZURIT Seniorenzentrum Erfurt   
8                   Deutschordens-Seniorenhaus Erfurt   
9   Diakoniestiftung Weimar - Seniorenzentrum Haus...   
10       DRK Senioren- und Pflegeheim Christianenheim   
11     DRK-Senioren- und Pflegeheim Albert Schweitzer   
12   Häusliche Kranken- und Altenpflege Schmid/Möller   
13                               Haus im Brühl Erfurt   
14                     HELIOS Residenz am Wiesenhügel   
15                        K&S Seniorenresidenz Erfurt   
16              Kath. Altenheim Carolinenstift Erfurt   
17                    Pflegewohnpark der Generationen   
18                           Residenz Ambiente Erfurt   
19       Senioren- und Pflegeheim Haus Am Steigerwald   
20                       Seniorenpflegeheim "Am Park"   
21           St.-Elisabeth-Heim Alten- und Pflegeheim   
22  VAMED Senioren- und Pflegeheim Erfurt "Haus am...   
23                                  Wohnen am Zoopark   

                                       addr  
0         Nordhäuser Straße 74,99089,Erfurt  
1            Haarbergstraße 72,99097,Erfurt  
2             Neuwerkstraße 24,99084,Erfurt  
3               Rankestraße 59,99096,Erfurt  
4            Hospitalplatz 15a,99084,Erfurt  
5      Heinrich-Heine-Straße 2,99096,Erfurt  
6        Juri-Gagarin-Ring 140,99084,Erfurt  
7      Theo-Neubauer-Straße 15,99085,Erfurt  
8          Vilniuser Straße 14,99089,Erfurt  
9      Große Ackerhofsgasse 14,99084,Erfurt  
10        Arnstädter Straße 48,99096,Erfurt  
11        Warschauer Straße 12,99089,Erfurt  
12    Stotternheimer Straße 9a,99086,Erfurt  
13      Placidus-Muth-Straße 2,99084,Erfurt  
14             Färberwaidweg 6,99097,Erfurt  
15  Maximilian-Welsch-Straße 5,99084,Erfurt  
16                  Pilse 9-11,99084,Erfurt  
17       Wermutmühlenweg 61-63,99089,Erfurt  
18          Häßlerstraße 19-21,99096,Erfurt  
19               Parkstraße 1a,99096,Erfurt  
20                  Amtsberg 4,99098,Erfurt  
21              Herderstraße 5,99096,Erfurt  
22             Baumerstraße 19,99089,Erfurt  
23       Julius-Leber-Ring 23a,99087,Erfurt

In [9]:
Nursing_and_care_home.shape

(24, 2)

In [10]:
from geopandas.tools import geocode
geo = geocode(Nursing_and_care_home['addr'], provider='nominatim', user_agent='autogis_xx', timeout=4)
geo

geometry  \
0   POINT (11.01345 50.99268)   
1   POINT (11.08423 50.94773)   
2   POINT (11.02681 50.97242)   
3   POINT (11.02884 50.96135)   
4   POINT (11.03666 50.98179)   
5   POINT (11.03266 50.96127)   
6   POINT (11.03541 50.98188)   
7   POINT (11.04643 50.98017)   
8   POINT (11.01086 51.00656)   
9   POINT (11.02152 50.98205)   
10  POINT (11.03392 50.96120)   
11  POINT (11.00716 51.00097)   
12  POINT (11.03272 51.00959)   
13  POINT (11.01666 50.97391)   
14  POINT (11.06700 50.95259)   
15  POINT (11.01678 50.97493)   
16  POINT (11.03203 50.97827)   
17  POINT (11.02118 50.99714)   
18  POINT (11.04084 50.96493)   
19  POINT (11.01794 50.96062)   
20  POINT (11.14716 50.99590)   
21  POINT (11.02977 50.96474)   
22  POINT (11.01598 50.98974)   
23  POINT (11.01992 51.01897)   

                                              address  
0   Helios Klinikum Erfurt, 74, Nordhäuser Straße,...  
1   Katholisches Krankenhaus "St. Johann Nepomuk",...  
2   Apotheke am Hirschgarten, 24, Neuwerkstraße, A...  
3   59, Rankestraße, Löbervorstadt, Erfurt, Thürin...  
4   Kita & Pflegeheim "Augusta-Viktoria-Stift", 15...  
5   2, Heinrich-Heine-Straße, Löbervorstadt, Erfur...  
6   Museum für Thüringer Volkskunde, 140 a, Juri-G...  
7   15, Theo-Neubauer-Straße, Äußere Krämpfervorst...  
8   Deutschordens-Seniorenhaus, 14, Vilniuser Stra...  
9   14, Große Ackerhofsgasse, Andreasviertel, Alts...  
10  48, Arnstädter Straße, Löbervorstadt, Erfurt, ...  
11  12, Warschauer Straße, Berliner Platz, Marbach...  
12  BKK der Thüringer Energieversorgung, 9a, Stott...  
13  Haus im Brühl, 2, Placidus-Muth-Straße, Erfurt...  
14  HELIOS Residenz am Wiesenhügel, 6, Färberwaidw...  
15  Seniorenresidenz, 5, Maximilian-Welsch-Straße,...  
16  Pilse, Altstadt, Erfurt, Thüringen, 99084, Deu...  
17  Wermutmühlenweg, Ilversgehofen, Erfurt, Thürin...  
18  Häßlerstraße, Daberstedt, Erfurt, Thüringen, 9...  
19  Senioren- und Pflegeheim Haus Am Steigerwald, ...  
20  Seniorenpflegeheim „Am Park“, 4, Amtsberg, Vie...  
21  St.-Elisabeth-Heim Alten- und Pflegeheim, 5, H...  
22  19, Baumerstraße, Andreasvorstadt, Erfurt, Thü...  
23  Am Zoopark, 23a, Julius-Leber-Ring, Roter Berg...

In [11]:
join_geo = geo.join(Nursing_and_care_home)
join_geo.head()

geometry  \
0  POINT (11.01345 50.99268)   
1  POINT (11.08423 50.94773)   
2  POINT (11.02681 50.97242)   
3  POINT (11.02884 50.96135)   
4  POINT (11.03666 50.98179)   

                                             address  \
0  Helios Klinikum Erfurt, 74, Nordhäuser Straße,...   
1  Katholisches Krankenhaus "St. Johann Nepomuk",...   
2  Apotheke am Hirschgarten, 24, Neuwerkstraße, A...   
3  59, Rankestraße, Löbervorstadt, Erfurt, Thürin...   
4  Kita & Pflegeheim "Augusta-Viktoria-Stift", 15...   

                                                name  \
0                             Helios Klinikum Erfurt   
1  Katholisches Krankenhaus "St. Johann Nepomuk" ...   
2       Alloheim Senioren-Residenz "Am Hirschgarten"   
3         ASB Senioren- und Pflegeheim "Georg Boock"   
4       Augusta-Viktoria-Stift Alten- und Pflegeheim   

                                addr  
0  Nordhäuser Straße 74,99089,Erfurt  
1     Haarbergstraße 72,99097,Erfurt  
2      Neuwerkstraße 24,99084,Erfurt  
3        Rankestraße 59,99096,Erfurt  
4     Hospitalplatz 15a,99084,Erfurt

In [12]:
#Importing the required modules
from geopy.extra.rate_limiter import RateLimiter
 
#Creating a dataframe with address of locations we want to reterive
locat = Nursing_and_care_home['addr']
df = pd.DataFrame({'add': locat})
 
#Creating an instance of Nominatim Class
geolocator = Nominatim(user_agent="my_request")
 
#applying the rate limiter wrapper
geocode = RateLimiter(geolocator.geocode, min_delay_seconds=1)
 
#Applying the method to pandas DataFrame
df['location'] = df['add'].apply(geocode)
df['Lat'] = df['location'].apply(lambda x: x.latitude if x else None)
df['Lon'] = df['location'].apply(lambda x: x.longitude if x else None)
df

add  \
0         Nordhäuser Straße 74,99089,Erfurt   
1            Haarbergstraße 72,99097,Erfurt   
2             Neuwerkstraße 24,99084,Erfurt   
3               Rankestraße 59,99096,Erfurt   
4            Hospitalplatz 15a,99084,Erfurt   
5      Heinrich-Heine-Straße 2,99096,Erfurt   
6        Juri-Gagarin-Ring 140,99084,Erfurt   
7      Theo-Neubauer-Straße 15,99085,Erfurt   
8          Vilniuser Straße 14,99089,Erfurt   
9      Große Ackerhofsgasse 14,99084,Erfurt   
10        Arnstädter Straße 48,99096,Erfurt   
11        Warschauer Straße 12,99089,Erfurt   
12    Stotternheimer Straße 9a,99086,Erfurt   
13      Placidus-Muth-Straße 2,99084,Erfurt   
14             Färberwaidweg 6,99097,Erfurt   
15  Maximilian-Welsch-Straße 5,99084,Erfurt   
16                  Pilse 9-11,99084,Erfurt   
17       Wermutmühlenweg 61-63,99089,Erfurt   
18          Häßlerstraße 19-21,99096,Erfurt   
19               Parkstraße 1a,99096,Erfurt   
20                  Amtsberg 4,99098,Erfurt   
21              Herderstraße 5,99096,Erfurt   
22             Baumerstraße 19,99089,Erfurt   
23       Julius-Leber-Ring 23a,99087,Erfurt   

                                             location        Lat        Lon  
0   (Helios Klinikum Erfurt, 74, Nordhäuser Straße...  50.992685  11.013446  
1   (Katholisches Krankenhaus "St. Johann Nepomuk"...  50.947735  11.084226  
2   (Apotheke am Hirschgarten, 24, Neuwerkstraße, ...  50.972417  11.026811  
3   (59, Rankestraße, Löbervorstadt, Erfurt, Thüri...  50.961353  11.028842  
4   (Kita & Pflegeheim "Augusta-Viktoria-Stift", 1...  50.981793  11.036660  
5   (2, Heinrich-Heine-Straße, Löbervorstadt, Erfu...  50.961271  11.032657  
6   (Museum für Thüringer Volkskunde, 140 a, Juri-...  50.981881  11.035409  
7   (15, Theo-Neubauer-Straße, Äußere Krämpfervors...  50.980172  11.046428  
8   (Deutschordens-Seniorenhaus, 14, Vilniuser Str...  51.006559  11.010863  
9   (14, Große Ackerhofsgasse, Andreasviertel, Alt...  50.982049  11.021522  
10  (48, Arnstädter Straße, Löbervorstadt, Erfurt,...  50.961200  11.033925  
11  (12, Warschauer Straße, Berliner Platz, Marbac...  51.000975  11.007161  
12  (BKK der Thüringer Energieversorgung, 9a, Stot...  51.009589  11.032718  
13  (Haus im Brühl, 2, Placidus-Muth-Straße, Erfur...  50.973914  11.016659  
14  (HELIOS Residenz am Wiesenhügel, 6, Färberwaid...  50.952592  11.066997  
15  (Seniorenresidenz, 5, Maximilian-Welsch-Straße...  50.974929  11.016781  
16  (Pilse, Altstadt, Erfurt, Thüringen, 99084, De...  50.978265  11.032025  
17  (Wermutmühlenweg, Ilversgehofen, Erfurt, Thüri...  50.997144  11.021180  
18  (Häßlerstraße, Daberstedt, Erfurt, Thüringen, ...  50.964932  11.040838  
19  (Senioren- und Pflegeheim Haus Am Steigerwald,...  50.960616  11.017943  
20  (Seniorenpflegeheim „Am Park“, 4, Amtsberg, Vi...  50.995896  11.147164  
21  (St.-Elisabeth-Heim Alten- und Pflegeheim, 5, ...  50.964740  11.029771  
22  (19, Baumerstraße, Andreasvorstadt, Erfurt, Th...  50.989737  11.015982  
23  (Am Zoopark, 23a, Julius-Leber-Ring, Roter Ber...  51.018974  11.019924

nominatim = Nominatim(user_agent="Erfurt")
coord = '50.9777974, 11.0287364'
#erfurt = nominatim.query(50.9777974,11.0287364, reverse=True, zoom=18)
#city = nominatim.query('Erfurt, Germany')
city['coord'] = coord
city

In [13]:
#df_LatLon = df.loc[:23, ['Lat', 'Lon']]
#df_LatLon
#nursing = df.join(join_geo)
#nursing.head()

In [14]:
#location = geolocator.reverse(nursing['Lat'][0], nursing['Lon'][0])
#results = Geocoder.reverse_geocode(df['latitude'][0], df['longitude'][0])
#print(location.address)
#nursing = nursing.astype({"Lat": float, "Lon": float})
#df['address'] = nursing.apply(lambda row: geolocator.reverse(row['Lat'],row['Lon']),axis=1)
#df
#nursing.dtypes

In [15]:
print(location.raw)

{'place_id': 257761667, 'licence': 'Data © OpenStreetMap contributors, ODbL 1.0. https://osm.org/copyright', 'osm_type': 'relation', 'osm_id': 62745, 'boundingbox': ['50.8904895', '51.079978', '10.8546679', '11.1751358'], 'lat': '50.9777974', 'lon': '11.0287364', 'display_name': 'Erfurt, Thüringen, Deutschland', 'class': 'boundary', 'type': 'administrative', 'importance': 0.7204711562778562, 'icon': 'https://nominatim.openstreetmap.org/ui/mapicons//poi_boundary_administrative.p.20.png'}


In [16]:
#df['latlon'] = nursing['Lat'].map(str) + ','+ nursing['Lon'].map(str)
df['latlon'] = df['Lat'].map(str) + ','+ df['Lon'].map(str)
df['latlon'][0]
df                                                       

add  \
0         Nordhäuser Straße 74,99089,Erfurt   
1            Haarbergstraße 72,99097,Erfurt   
2             Neuwerkstraße 24,99084,Erfurt   
3               Rankestraße 59,99096,Erfurt   
4            Hospitalplatz 15a,99084,Erfurt   
5      Heinrich-Heine-Straße 2,99096,Erfurt   
6        Juri-Gagarin-Ring 140,99084,Erfurt   
7      Theo-Neubauer-Straße 15,99085,Erfurt   
8          Vilniuser Straße 14,99089,Erfurt   
9      Große Ackerhofsgasse 14,99084,Erfurt   
10        Arnstädter Straße 48,99096,Erfurt   
11        Warschauer Straße 12,99089,Erfurt   
12    Stotternheimer Straße 9a,99086,Erfurt   
13      Placidus-Muth-Straße 2,99084,Erfurt   
14             Färberwaidweg 6,99097,Erfurt   
15  Maximilian-Welsch-Straße 5,99084,Erfurt   
16                  Pilse 9-11,99084,Erfurt   
17       Wermutmühlenweg 61-63,99089,Erfurt   
18          Häßlerstraße 19-21,99096,Erfurt   
19               Parkstraße 1a,99096,Erfurt   
20                  Amtsberg 4,99098,Erfurt   
21              Herderstraße 5,99096,Erfurt   
22             Baumerstraße 19,99089,Erfurt   
23       Julius-Leber-Ring 23a,99087,Erfurt   

                                             location        Lat        Lon  \
0   (Helios Klinikum Erfurt, 74, Nordhäuser Straße...  50.992685  11.013446   
1   (Katholisches Krankenhaus "St. Johann Nepomuk"...  50.947735  11.084226   
2   (Apotheke am Hirschgarten, 24, Neuwerkstraße, ...  50.972417  11.026811   
3   (59, Rankestraße, Löbervorstadt, Erfurt, Thüri...  50.961353  11.028842   
4   (Kita & Pflegeheim "Augusta-Viktoria-Stift", 1...  50.981793  11.036660   
5   (2, Heinrich-Heine-Straße, Löbervorstadt, Erfu...  50.961271  11.032657   
6   (Museum für Thüringer Volkskunde, 140 a, Juri-...  50.981881  11.035409   
7   (15, Theo-Neubauer-Straße, Äußere Krämpfervors...  50.980172  11.046428   
8   (Deutschordens-Seniorenhaus, 14, Vilniuser Str...  51.006559  11.010863   
9   (14, Große Ackerhofsgasse, Andreasviertel, Alt...  50.982049  11.021522   
10  (48, Arnstädter Straße, Löbervorstadt, Erfurt,...  50.961200  11.033925   
11  (12, Warschauer Straße, Berliner Platz, Marbac...  51.000975  11.007161   
12  (BKK der Thüringer Energieversorgung, 9a, Stot...  51.009589  11.032718   
13  (Haus im Brühl, 2, Placidus-Muth-Straße, Erfur...  50.973914  11.016659   
14  (HELIOS Residenz am Wiesenhügel, 6, Färberwaid...  50.952592  11.066997   
15  (Seniorenresidenz, 5, Maximilian-Welsch-Straße...  50.974929  11.016781   
16  (Pilse, Altstadt, Erfurt, Thüringen, 99084, De...  50.978265  11.032025   
17  (Wermutmühlenweg, Ilversgehofen, Erfurt, Thüri...  50.997144  11.021180   
18  (Häßlerstraße, Daberstedt, Erfurt, Thüringen, ...  50.964932  11.040838   
19  (Senioren- und Pflegeheim Haus Am Steigerwald,...  50.960616  11.017943   
20  (Seniorenpflegeheim „Am Park“, 4, Amtsberg, Vi...  50.995896  11.147164   
21  (St.-Elisabeth-Heim Alten- und Pflegeheim, 5, ...  50.964740  11.029771   
22  (19, Baumerstraße, Andreasvorstadt, Erfurt, Th...  50.989737  11.015982   
23  (Am Zoopark, 23a, Julius-Leber-Ring, Roter Ber...  51.018974  11.019924   

                                   latlon  
0           50.99268485,11.01344586977623  
1          50.94773485,11.084226172724772  
2                    50.972417,11.0268113  
3          50.96135325,11.028842394954673  
4           50.9817928,11.036660451209475  
5                   50.9612709,11.0326575  
6            50.9818814,11.03540916526946  
7                   50.9801717,11.0464282  
8   51.006558749999996,11.010862517930299  
9                     50.982049,11.021522  
10                  50.9612004,11.0339249  
11                  51.0009746,11.0071613  
12                   51.0095886,11.032718  
13                   50.973914,11.0166587  
14  50.952592249999995,11.066997174512938  
15         50.97492895,11.016781446960117  
16                  50.9782655,11.0320254  
17                  50.9971437,11.0211796  
18                  50.9649324,11.0408382  
19      

In [17]:
#We create a service provider locator this time pass timeout to be 10 minutes since we have many rows. That will prevent timeout errors that you might encounter during the process
locator = Nominatim(user_agent='myGeocoder', timeout=10)
rgeocode = RateLimiter(locator.reverse, min_delay_seconds=0.001)
print('done')

done


In [18]:
#Now, we can reverse geocode by using applying the reverse geocoding rgeocode we created above. To get a nice progress bar, we are using here tqdm library.
#location = geolocator.reverse(df['latlon'][0])
#print(location.address)

In [19]:
#df['gesamt'] = None
#location = geolocator.reverse(df['latlon'][7])
#df['gesamt']= location.address
#df

In [20]:
df['full_adress'] = None
i = 0
while i < 24:
    location = geolocator.reverse(df['latlon'][i])
    #print(location.address)
    df.loc[i,['full_adress']]= location.address
    i += 1
    #df['full_adress']= location.address
df

add  \
0         Nordhäuser Straße 74,99089,Erfurt   
1            Haarbergstraße 72,99097,Erfurt   
2             Neuwerkstraße 24,99084,Erfurt   
3               Rankestraße 59,99096,Erfurt   
4            Hospitalplatz 15a,99084,Erfurt   
5      Heinrich-Heine-Straße 2,99096,Erfurt   
6        Juri-Gagarin-Ring 140,99084,Erfurt   
7      Theo-Neubauer-Straße 15,99085,Erfurt   
8          Vilniuser Straße 14,99089,Erfurt   
9      Große Ackerhofsgasse 14,99084,Erfurt   
10        Arnstädter Straße 48,99096,Erfurt   
11        Warschauer Straße 12,99089,Erfurt   
12    Stotternheimer Straße 9a,99086,Erfurt   
13      Placidus-Muth-Straße 2,99084,Erfurt   
14             Färberwaidweg 6,99097,Erfurt   
15  Maximilian-Welsch-Straße 5,99084,Erfurt   
16                  Pilse 9-11,99084,Erfurt   
17       Wermutmühlenweg 61-63,99089,Erfurt   
18          Häßlerstraße 19-21,99096,Erfurt   
19               Parkstraße 1a,99096,Erfurt   
20                  Amtsberg 4,99098,Erfurt   
21              Herderstraße 5,99096,Erfurt   
22             Baumerstraße 19,99089,Erfurt   
23       Julius-Leber-Ring 23a,99087,Erfurt   

                                             location        Lat        Lon  \
0   (Helios Klinikum Erfurt, 74, Nordhäuser Straße...  50.992685  11.013446   
1   (Katholisches Krankenhaus "St. Johann Nepomuk"...  50.947735  11.084226   
2   (Apotheke am Hirschgarten, 24, Neuwerkstraße, ...  50.972417  11.026811   
3   (59, Rankestraße, Löbervorstadt, Erfurt, Thüri...  50.961353  11.028842   
4   (Kita & Pflegeheim "Augusta-Viktoria-Stift", 1...  50.981793  11.036660   
5   (2, Heinrich-Heine-Straße, Löbervorstadt, Erfu...  50.961271  11.032657   
6   (Museum für Thüringer Volkskunde, 140 a, Juri-...  50.981881  11.035409   
7   (15, Theo-Neubauer-Straße, Äußere Krämpfervors...  50.980172  11.046428   
8   (Deutschordens-Seniorenhaus, 14, Vilniuser Str...  51.006559  11.010863   
9   (14, Große Ackerhofsgasse, Andreasviertel, Alt...  50.982049  11.021522   
10  (48, Arnstädter Straße, Löbervorstadt, Erfurt,...  50.961200  11.033925   
11  (12, Warschauer Straße, Berliner Platz, Marbac...  51.000975  11.007161   
12  (BKK der Thüringer Energieversorgung, 9a, Stot...  51.009589  11.032718   
13  (Haus im Brühl, 2, Placidus-Muth-Straße, Erfur...  50.973914  11.016659   
14  (HELIOS Residenz am Wiesenhügel, 6, Färberwaid...  50.952592  11.066997   
15  (Seniorenresidenz, 5, Maximilian-Welsch-Straße...  50.974929  11.016781   
16  (Pilse, Altstadt, Erfurt, Thüringen, 99084, De...  50.978265  11.032025   
17  (Wermutmühlenweg, Ilversgehofen, Erfurt, Thüri...  50.997144  11.021180   
18  (Häßlerstraße, Daberstedt, Erfurt, Thüringen, ...  50.964932  11.040838   
19  (Senioren- und Pflegeheim Haus Am Steigerwald,...  50.960616  11.017943   
20  (Seniorenpflegeheim „Am Park“, 4, Amtsberg, Vi...  50.995896  11.147164   
21  (St.-Elisabeth-Heim Alten- und Pflegeheim, 5, ...  50.964740  11.029771   
22  (19, Baumerstraße, Andreasvorstadt, Erfurt, Th...  50.989737  11.015982   
23  (Am Zoopark, 23a, Julius-Leber-Ring, Roter Ber...  51.018974  11.019924   

                                   latlon  \
0           50.99268485,11.01344586977623   
1          50.94773485,11.084226172724772   
2                    50.972417,11.0268113   
3          50.96135325,11.028842394954673   
4           50.9817928,11.036660451209475   
5                   50.9612709,11.0326575   
6            50.9818814,11.03540916526946   
7                   50.9801717,11.0464282   
8   51.006558749999996,11.010862517930299   
9                     50.982049,11.021522   
10                  50.9612004,11.0339249   
11                  51.0009746,11.0071613   
12                   51.0095886,11.032718   
13                   50.973914,11.0166587   
14  50.952592249999995,11.066997174512938   
15         50.97492895,11.016781446960117   
16                  50.9782655,11.0320254   
17                  50.9971437,11.0211796   
18                  50.9649324,11

In [21]:
#df['Name'], df['HSNR'], df['street'], df['suburb'], df['city'],df['state'], df['PLZ'], df['country'] = df["full_adress"].str.split(",").str
df[['Name','HSNR','street','suburb','city','state','PLZ','country']] = df['full_adress'].str.split(',', 7, expand=True)
df =df.drop(['HSNR','add', 'location', 'state','PLZ','country','Name'],axis=1)
df

Lat        Lon                                 latlon  \
0   50.992685  11.013446          50.99268485,11.01344586977623   
1   50.947735  11.084226         50.94773485,11.084226172724772   
2   50.972417  11.026811                   50.972417,11.0268113   
3   50.961353  11.028842         50.96135325,11.028842394954673   
4   50.981793  11.036660          50.9817928,11.036660451209475   
5   50.961271  11.032657                  50.9612709,11.0326575   
6   50.981881  11.035409           50.9818814,11.03540916526946   
7   50.980172  11.046428                  50.9801717,11.0464282   
8   51.006559  11.010863  51.006558749999996,11.010862517930299   
9   50.982049  11.021522                    50.982049,11.021522   
10  50.961200  11.033925                  50.9612004,11.0339249   
11  51.000975  11.007161                  51.0009746,11.0071613   
12  51.009589  11.032718                   51.0095886,11.032718   
13  50.973914  11.016659                   50.973914,11.0166587   
14  50.952592  11.066997  50.952592249999995,11.066997174512938   
15  50.974929  11.016781         50.97492895,11.016781446960117   
16  50.978265  11.032025                  50.9782655,11.0320254   
17  50.997144  11.021180                  50.9971437,11.0211796   
18  50.964932  11.040838                  50.9649324,11.0408382   
19  50.960616  11.017943           50.9606163,11.01794267217284   
20  50.995896  11.147164  50.995895649999994,11.147163903392796   
21  50.964740  11.029771         50.96474015,11.029770865071772   
22  50.989737  11.015982                   50.989737,11.0159822   
23  51.018974  11.019924          51.0189736,11.019923862956109   

                                          full_adress  \
0   Helios Klinikum Erfurt, 74, Nordhäuser Straße,...   
1   Katholisches Krankenhaus "St. Johann Nepomuk",...   
2   Apotheke am Hirschgarten, 24, Neuwerkstraße, A...   
3   ASB- Seniorenheim "Georg Book", Rankestraße, L...   
4   Kita & Pflegeheim "Augusta-Viktoria-Stift", 15...   
5   2, Heinrich-Heine-Straße, Löbervorstadt, Erfur...   
6   Großes Hospital, Juri-Gagarin-Ring, Altstadt, ...   
7   15, Theo-Neubauer-Straße, Äußere Krämpfervorst...   
8   Deutschordens-Seniorenhaus, 14, Vilniuser Stra...   
9   14, Große Ackerhofsgasse, Andreasviertel, Alts...   
10  48, Arnstädter Straße, Löbervorstadt, Erfurt, ...   
11  12, Warschauer Straße, Berliner Platz, Marbach...   
12  9a, Stotternheimer Straße, Jacobsenviertel, Il...   
13  Haus im Brühl, 2, Placidus-Muth-Straße, Erfurt...   
14  HELIOS Residenz am Wiesenhügel, 6, Färberwaidw...   
15  Seniorenresidenz, 5, Maximilian-Welsch-Straße,...   
16  Zur Kemmlatte, Pilse, Altstadt, Erfurt, Thürin...   
17  11, Wermutmühlenweg, Ilversgehofen, Erfurt, Th...   
18  Häßlerstraße, Daberstedt, Erfurt, Thüringen, 9...   
19  Senioren- und Pflegeheim Haus Am Steigerwald, ...   
20  Seniorenpflegeheim „Am Park“, 4, Amtsberg, Vie...   
21  St.-Elisabeth-Heim Alten- und Pflegeheim, 5, H...   
22  Helios Klinikum Erfurt, 74, Nordhäuser Straße,...   
23  Am Zoopark, 23a, Julius-Leber-Ring, Roter Berg...   

                       street             suburb              city  
0           Nordhäuser Straße    Andreasvorstadt            Erfurt  
1              Haarbergstraße        Melchendorf            Urbich  
2               Neuwerkstraße           Altstadt            Erfurt  
3               Löbervorstadt             Erfurt         Thüringen  
4               Hospitalplatz           Altstadt            Erfurt  
5               Löbervorstadt             Erfurt         Thüringen  
6                    Altstadt             Erfurt         Thüringen  
7     Äußere Krämpfervorstadt   Krämpfervorstadt            Erfurt  
8            Vilniuser Straße              Rieth            Erfurt  
9              Andreasviertel           Altstadt            Erfurt  
10              Löbervorstadt             Erfurt         Thüringen  
11             Berliner Platz            Marbach            Erfurt  
12         

In [22]:
Nursing_and_care_home_new = df.join(Nursing_and_care_home)
#Nursing_and_care_home_new = Nursing_and_care_home_new.drop(['addr'],axis=1)
Nursing_and_care_home_new.head()

Lat        Lon                          latlon  \
0  50.992685  11.013446   50.99268485,11.01344586977623   
1  50.947735  11.084226  50.94773485,11.084226172724772   
2  50.972417  11.026811            50.972417,11.0268113   
3  50.961353  11.028842  50.96135325,11.028842394954673   
4  50.981793  11.036660   50.9817928,11.036660451209475   

                                         full_adress              street  \
0  Helios Klinikum Erfurt, 74, Nordhäuser Straße,...   Nordhäuser Straße   
1  Katholisches Krankenhaus "St. Johann Nepomuk",...      Haarbergstraße   
2  Apotheke am Hirschgarten, 24, Neuwerkstraße, A...       Neuwerkstraße   
3  ASB- Seniorenheim "Georg Book", Rankestraße, L...       Löbervorstadt   
4  Kita & Pflegeheim "Augusta-Viktoria-Stift", 15...       Hospitalplatz   

             suburb        city  \
0   Andreasvorstadt      Erfurt   
1       Melchendorf      Urbich   
2          Altstadt      Erfurt   
3            Erfurt   Thüringen   
4          Altstadt      Erfurt   

                                                name  \
0                             Helios Klinikum Erfurt   
1  Katholisches Krankenhaus "St. Johann Nepomuk" ...   
2       Alloheim Senioren-Residenz "Am Hirschgarten"   
3         ASB Senioren- und Pflegeheim "Georg Boock"   
4       Augusta-Viktoria-Stift Alten- und Pflegeheim   

                                addr  
0  Nordhäuser Straße 74,99089,Erfurt  
1     Haarbergstraße 72,99097,Erfurt  
2      Neuwerkstraße 24,99084,Erfurt  
3        Rankestraße 59,99096,Erfurt  
4     Hospitalplatz 15a,99084,Erfurt

In [23]:
Nursing_and_care_home_new = Nursing_and_care_home_new.drop(['full_adress','addr'],axis=1)
Nursing_and_care_home_new = Nursing_and_care_home_new[['name','Lat','Lon','latlon','street','suburb','city']]
Nursing_and_care_home_new

name        Lat        Lon  \
0                              Helios Klinikum Erfurt  50.992685  11.013446   
1   Katholisches Krankenhaus "St. Johann Nepomuk" ...  50.947735  11.084226   
2        Alloheim Senioren-Residenz "Am Hirschgarten"  50.972417  11.026811   
3          ASB Senioren- und Pflegeheim "Georg Boock"  50.961353  11.028842   
4        Augusta-Viktoria-Stift Alten- und Pflegeheim  50.981793  11.036660   
5                   Augusta-Viktoria-Stift Pflegeheim  50.961271  11.032657   
6   AWO-Altenpflegeheim Haus zu den vier Jahreszeiten  50.981881  11.035409   
7                       AZURIT Seniorenzentrum Erfurt  50.980172  11.046428   
8                   Deutschordens-Seniorenhaus Erfurt  51.006559  11.010863   
9   Diakoniestiftung Weimar - Seniorenzentrum Haus...  50.982049  11.021522   
10       DRK Senioren- und Pflegeheim Christianenheim  50.961200  11.033925   
11     DRK-Senioren- und Pflegeheim Albert Schweitzer  51.000975  11.007161   
12   Häusliche Kranken- und Altenpflege Schmid/Möller  51.009589  11.032718   
13                               Haus im Brühl Erfurt  50.973914  11.016659   
14                     HELIOS Residenz am Wiesenhügel  50.952592  11.066997   
15                        K&S Seniorenresidenz Erfurt  50.974929  11.016781   
16              Kath. Altenheim Carolinenstift Erfurt  50.978265  11.032025   
17                    Pflegewohnpark der Generationen  50.997144  11.021180   
18                           Residenz Ambiente Erfurt  50.964932  11.040838   
19       Senioren- und Pflegeheim Haus Am Steigerwald  50.960616  11.017943   
20                       Seniorenpflegeheim "Am Park"  50.995896  11.147164   
21           St.-Elisabeth-Heim Alten- und Pflegeheim  50.964740  11.029771   
22  VAMED Senioren- und Pflegeheim Erfurt "Haus am...  50.989737  11.015982   
23                                  Wohnen am Zoopark  51.018974  11.019924   

                                   latlon                     street  \
0           50.99268485,11.01344586977623          Nordhäuser Straße   
1          50.94773485,11.084226172724772             Haarbergstraße   
2                    50.972417,11.0268113              Neuwerkstraße   
3          50.96135325,11.028842394954673              Löbervorstadt   
4           50.9817928,11.036660451209475              Hospitalplatz   
5                   50.9612709,11.0326575              Löbervorstadt   
6            50.9818814,11.03540916526946                   Altstadt   
7                   50.9801717,11.0464282    Äußere Krämpfervorstadt   
8   51.006558749999996,11.010862517930299           Vilniuser Straße   
9                     50.982049,11.021522             Andreasviertel   
10                  50.9612004,11.0339249              Löbervorstadt   
11                  51.0009746,11.0071613             Berliner Platz   
12                   51.0095886,11.032718            Jacobsenviertel   
13                   50.973914,11.0166587       Placidus-Muth-Straße   
14  50.952592249999995,11.066997174512938              Färberwaidweg   
15         50.97492895,11.016781446960117   Maximilian-Welsch-Straße   
16                  50.9782655,11.0320254                   Altstadt   
17                  50.9971437,11.0211796              Ilversgehofen   
18                  50.9649324,11.0408382                     Erfurt   
19           50.9606163,11.01794267217284                 Parkstraße   
20  50.995895649999994,11.147163903392796                   Amtsberg   
21         50.96474015,11.029770865071772               Herderstraße   
22                   50.989737,11.0159822          Nordhäuser Straße   
23          51.0189736,11.019923862956109          Julius-Leber-Ring   

               suburb              city  
0     Andreasvorstadt            Erfurt  
1         Melchendorf            Urbich  
2            Altstadt            Erfurt  
3              Erfurt         Thüringen  
4            Altstadt            Erfurt  
5              Erfur

In [24]:
#Nursing_and_care_home_new.loc[[3,5,10'],'suburb']replace['Erfurt':'Löbervorstadt']
                               
#where df['suburb'] == 'Erfurt' and df['suburb'] == 'Erfurt' 
 #   set df['suburb'] to = 'Löbervorstadt'
#Nursing_and_care_home_new

In [25]:
#df.loc[df.Weight == "155", "Name"] = "John"
#Nursing_and_care_home_new.loc[Nursing_and_care_home_new.street=="Löbervorstadt","suburb"] = "Löbervorstadt"
Nursing_and_care_home_new.loc[[3,5,10], ['suburb']] = "Löbervorstadt"
Nursing_and_care_home_new.loc[[6,16], ['suburb']] = "Altstadt"
Nursing_and_care_home_new.loc[[13,15], ['suburb']] = "Brühlervorstadt"
Nursing_and_care_home_new.loc[[17], ['suburb']] = "Ilversgehofen"
Nursing_and_care_home_new.loc[[18], ['suburb']] = 'Daberstedt'
Nursing_and_care_home_new = Nursing_and_care_home_new.drop(['city'],axis=1)
Nursing_and_care_home_new

name        Lat        Lon  \
0                              Helios Klinikum Erfurt  50.992685  11.013446   
1   Katholisches Krankenhaus "St. Johann Nepomuk" ...  50.947735  11.084226   
2        Alloheim Senioren-Residenz "Am Hirschgarten"  50.972417  11.026811   
3          ASB Senioren- und Pflegeheim "Georg Boock"  50.961353  11.028842   
4        Augusta-Viktoria-Stift Alten- und Pflegeheim  50.981793  11.036660   
5                   Augusta-Viktoria-Stift Pflegeheim  50.961271  11.032657   
6   AWO-Altenpflegeheim Haus zu den vier Jahreszeiten  50.981881  11.035409   
7                       AZURIT Seniorenzentrum Erfurt  50.980172  11.046428   
8                   Deutschordens-Seniorenhaus Erfurt  51.006559  11.010863   
9   Diakoniestiftung Weimar - Seniorenzentrum Haus...  50.982049  11.021522   
10       DRK Senioren- und Pflegeheim Christianenheim  50.961200  11.033925   
11     DRK-Senioren- und Pflegeheim Albert Schweitzer  51.000975  11.007161   
12   Häusliche Kranken- und Altenpflege Schmid/Möller  51.009589  11.032718   
13                               Haus im Brühl Erfurt  50.973914  11.016659   
14                     HELIOS Residenz am Wiesenhügel  50.952592  11.066997   
15                        K&S Seniorenresidenz Erfurt  50.974929  11.016781   
16              Kath. Altenheim Carolinenstift Erfurt  50.978265  11.032025   
17                    Pflegewohnpark der Generationen  50.997144  11.021180   
18                           Residenz Ambiente Erfurt  50.964932  11.040838   
19       Senioren- und Pflegeheim Haus Am Steigerwald  50.960616  11.017943   
20                       Seniorenpflegeheim "Am Park"  50.995896  11.147164   
21           St.-Elisabeth-Heim Alten- und Pflegeheim  50.964740  11.029771   
22  VAMED Senioren- und Pflegeheim Erfurt "Haus am...  50.989737  11.015982   
23                                  Wohnen am Zoopark  51.018974  11.019924   

                                   latlon                     street  \
0           50.99268485,11.01344586977623          Nordhäuser Straße   
1          50.94773485,11.084226172724772             Haarbergstraße   
2                    50.972417,11.0268113              Neuwerkstraße   
3          50.96135325,11.028842394954673              Löbervorstadt   
4           50.9817928,11.036660451209475              Hospitalplatz   
5                   50.9612709,11.0326575              Löbervorstadt   
6            50.9818814,11.03540916526946                   Altstadt   
7                   50.9801717,11.0464282    Äußere Krämpfervorstadt   
8   51.006558749999996,11.010862517930299           Vilniuser Straße   
9                     50.982049,11.021522             Andreasviertel   
10                  50.9612004,11.0339249              Löbervorstadt   
11                  51.0009746,11.0071613             Berliner Platz   
12                   51.0095886,11.032718            Jacobsenviertel   
13                   50.973914,11.0166587       Placidus-Muth-Straße   
14  50.952592249999995,11.066997174512938              Färberwaidweg   
15         50.97492895,11.016781446960117   Maximilian-Welsch-Straße   
16                  50.9782655,11.0320254                   Altstadt   
17                  50.9971437,11.0211796              Ilversgehofen   
18                  50.9649324,11.0408382                     Erfurt   
19           50.9606163,11.01794267217284                 Parkstraße   
20  50.995895649999994,11.147163903392796                   Amtsberg   
21         50.96474015,11.029770865071772               Herderstraße   
22                   50.989737,11.0159822          Nordhäuser Straße   
23          51.0189736,11.019923862956109          Julius-Leber-Ring   

               suburb  
0     Andreasvorstadt  
1         Melchendorf  
2            Altstadt  
3       Löbervorstadt  
4            Altstadt  
5       Löbervorstadt  
6            Altstadt  
7    Krämpfervorstadt  
8               Rieth  
9            Altstadt  
10      

In [26]:
Nursing_and_care_home_new = Nursing_and_care_home_new.drop(['street'],axis=1)


In [28]:
Nursing_and_care_home_new.set_index('suburb', inplace = True)
Nursing_and_care_home_new.sort_values(by='suburb', inplace=True)
Nursing_and_care_home_new

name  \
suburb                                                                 
 Altstadt               Alloheim Senioren-Residenz "Am Hirschgarten"   
 Altstadt               Augusta-Viktoria-Stift Alten- und Pflegeheim   
 Altstadt          Diakoniestiftung Weimar - Seniorenzentrum Haus...   
 Andreasvorstadt                              Helios Klinikum Erfurt   
 Andreasvorstadt   VAMED Senioren- und Pflegeheim Erfurt "Haus am...   
 Ilversgehofen      Häusliche Kranken- und Altenpflege Schmid/Möller   
 Krämpfervorstadt                      AZURIT Seniorenzentrum Erfurt   
 Löbervorstadt              St.-Elisabeth-Heim Alten- und Pflegeheim   
 Löbervorstadt          Senioren- und Pflegeheim Haus Am Steigerwald   
 Marbach              DRK-Senioren- und Pflegeheim Albert Schweitzer   
 Melchendorf       Katholisches Krankenhaus "St. Johann Nepomuk" ...   
 Rieth                             Deutschordens-Seniorenhaus Erfurt   
 Roter Berg                                        Wohnen am Zoopark   
 Vieselbach                             Seniorenpflegeheim "Am Park"   
 Wiesenhügel                          HELIOS Residenz am Wiesenhügel   
Altstadt           AWO-Altenpflegeheim Haus zu den vier Jahreszeiten   
Altstadt                       Kath. Altenheim Carolinenstift Erfurt   
Brühlervorstadt                                 Haus im Brühl Erfurt   
Brühlervorstadt                          K&S Seniorenresidenz Erfurt   
Daberstedt                                  Residenz Ambiente Erfurt   
Ilversgehofen                        Pflegewohnpark der Generationen   
Löbervorstadt           DRK Senioren- und Pflegeheim Christianenheim   
Löbervorstadt                      Augusta-Viktoria-Stift Pflegeheim   
Löbervorstadt             ASB Senioren- und Pflegeheim "Georg Boock"   

                         Lat        Lon                                 latlon  
suburb                                                                          
 Altstadt          50.972417  11.026811                   50.972417,11.0268113  
 Altstadt          50.981793  11.036660          50.9817928,11.036660451209475  
 Altstadt          50.982049  11.021522                    50.982049,11.021522  
 Andreasvorstadt   50.992685  11.013446          50.99268485,11.01344586977623  
 Andreasvorstadt   50.989737  11.015982                   50.989737,11.0159822  
 Ilversgehofen     51.009589  11.032718                   51.0095886,11.032718  
 Krämpfervorstadt  50.980172  11.046428                  50.9801717,11.0464282  
 Löbervorstadt     50.964740  11.029771         50.96474015,11.029770865071772  
 Löbervorstadt     50.960616  11.017943           50.9606163,11.01794267217284  
 Marbach           51.000975  11.007161                  51.0009746,11.0071613  
 Melchendorf       50.947735  11.084226         50.94773485,11.084226172724772  
 Rieth             51.006559  11.010863  51.006558749999996,11.010862517930299  
 Roter Berg        51.018974  11.019924          51.0189736,11.019923862956109  
 Vieselbach        50.995896  11.147164  50.995895649999994,11.147163903392796  
 Wiesenhügel       50.952592  11.066997  50.952592249999995,11.066997174512938  
Altstadt           50.981881  11.035409           50.9818814,11.03540916526946  
Altstadt           50.978265  11.032025                  50.9782655,11.0320254  
Brühlervorstadt    50.973914  11.016659                   50.973914,11.0166587  
Brühlervorstadt    50.974929  11.016781         50.97492895,11.016781446960117  
Daberstedt         50.964932  11.040838                  50.9649324,11.0408382  
Ilversgehofen      50.997144  11.021180                  50.9971437,11.0211796  
Löbervorstadt      50.961200  11.033925                  50.9612004,11.0339249  
Löbervorstadt      50.961271  11.032657                  50.9612709,11.0326575  
Löbervorstadt      50.961353  11.028842         50.96135325,11.028842394954673

In [33]:
Nurse_Care = Nursing_and_care_home_new.sort_values(by='suburb',ascending=False)
Nurse_Care

name  \
suburb                                                                 
Löbervorstadt             ASB Senioren- und Pflegeheim "Georg Boock"   
Löbervorstadt                      Augusta-Viktoria-Stift Pflegeheim   
Löbervorstadt           DRK Senioren- und Pflegeheim Christianenheim   
Ilversgehofen                        Pflegewohnpark der Generationen   
Daberstedt                                  Residenz Ambiente Erfurt   
Brühlervorstadt                          K&S Seniorenresidenz Erfurt   
Brühlervorstadt                                 Haus im Brühl Erfurt   
Altstadt                       Kath. Altenheim Carolinenstift Erfurt   
Altstadt           AWO-Altenpflegeheim Haus zu den vier Jahreszeiten   
 Wiesenhügel                          HELIOS Residenz am Wiesenhügel   
 Vieselbach                             Seniorenpflegeheim "Am Park"   
 Roter Berg                                        Wohnen am Zoopark   
 Rieth                             Deutschordens-Seniorenhaus Erfurt   
 Melchendorf       Katholisches Krankenhaus "St. Johann Nepomuk" ...   
 Marbach              DRK-Senioren- und Pflegeheim Albert Schweitzer   
 Löbervorstadt          Senioren- und Pflegeheim Haus Am Steigerwald   
 Löbervorstadt              St.-Elisabeth-Heim Alten- und Pflegeheim   
 Krämpfervorstadt                      AZURIT Seniorenzentrum Erfurt   
 Ilversgehofen      Häusliche Kranken- und Altenpflege Schmid/Möller   
 Andreasvorstadt   VAMED Senioren- und Pflegeheim Erfurt "Haus am...   
 Andreasvorstadt                              Helios Klinikum Erfurt   
 Altstadt               Augusta-Viktoria-Stift Alten- und Pflegeheim   
 Altstadt          Diakoniestiftung Weimar - Seniorenzentrum Haus...   
 Altstadt               Alloheim Senioren-Residenz "Am Hirschgarten"   

                         Lat        Lon                                 latlon  
suburb                                                                          
Löbervorstadt      50.961353  11.028842         50.96135325,11.028842394954673  
Löbervorstadt      50.961271  11.032657                  50.9612709,11.0326575  
Löbervorstadt      50.961200  11.033925                  50.9612004,11.0339249  
Ilversgehofen      50.997144  11.021180                  50.9971437,11.0211796  
Daberstedt         50.964932  11.040838                  50.9649324,11.0408382  
Brühlervorstadt    50.974929  11.016781         50.97492895,11.016781446960117  
Brühlervorstadt    50.973914  11.016659                   50.973914,11.0166587  
Altstadt           50.978265  11.032025                  50.9782655,11.0320254  
Altstadt           50.981881  11.035409           50.9818814,11.03540916526946  
 Wiesenhügel       50.952592  11.066997  50.952592249999995,11.066997174512938  
 Vieselbach        50.995896  11.147164  50.995895649999994,11.147163903392796  
 Roter Berg        51.018974  11.019924          51.0189736,11.019923862956109  
 Rieth             51.006559  11.010863  51.006558749999996,11.010862517930299  
 Melchendorf       50.947735  11.084226         50.94773485,11.084226172724772  
 Marbach           51.000975  11.007161                  51.0009746,11.0071613  
 Löbervorstadt     50.960616  11.017943           50.9606163,11.01794267217284  
 Löbervorstadt     50.964740  11.029771         50.96474015,11.029770865071772  
 Krämpfervorstadt  50.980172  11.046428                  50.9801717,11.0464282  
 Ilversgehofen     51.009589  11.032718                   51.0095886,11.032718  
 Andreasvorstadt   50.989737  11.015982                   50.989737,11.0159822  
 Andreasvorstadt   50.992685  11.013446          50.99268485,11.01344586977623  
 Altstadt          50.981793  11.036660          50.9817928,11.036660451209475  
 Altstadt          50.982049  11.021522                    50.982049,11.021522  
 Altstadt          50.972417  11.026811                   50.972417,11.0268113